eventuate-core/src/test/scala/com/rbmhtechnology/eventuate/EventsourcedActorTestKit.scala

```diff
@@ -0,0 +1,295 @@
+
+object EventSourcedActorTestKit {
+
+  /** actor system all actors under test run in */
+  val actorSystem = ActorSystem("test", ConfigFactory.parseString("eventuate.log.write-timeout = 1s"))
+
+  /** Simulate persisting 'event'. Actors this is sent to must implement [[ExplicitOnEventCall]] */
+  case class Persist(event: Any)
+
+}
+
+/**
+  * Actor tests need to implement this trait as well as [[akka.testkit.TestKit]]. A default [[ActorSystem]] is given
+  * by companion object.  Example for usage of this trait: <br>
+  *
+  * {{{ class MyActorTest extends TestKit(ActorTest.actorSystem)  with ActorTest with GivenWhenThen { ... } }}}
+  *
+  * Usage: <br>
+  * - instantiate actors to test via initEventSourcedActorAsTestActorRef or initEventSourcedActor. Former returns an
+  * [[TestActorRef]] and thus allows to test _.underlyingActor. <br>
+  * - either tell returned [[ActorRef]]/[[TestActorRef]] a command or explicitly call onEvent via #callOnEvent <br>
+  * - query 'peristed' events via #expectEvent or #expectEvents <br><br>
+  *
+  * All actor's under test must have #eventLog as their eventLog. Provided event log can be viewed as a stash. Events
+  * persisted by actors using provided eventLog are stashed. By calling #expectEvent or #expectEvents one or several
+  * events are popped from stash while asserting equality to user defined output. <br>
+  * By calling #cleanEventLog the whole stash can be cleaned. <br>
+  * If eventLogs content should not be asserted, but simply queried #queryEventlog can be used. Cleans the whole stash
+  * as well.
+  *
+  * fixme: move to reasonable package.
+  */
+trait EventSourcedActorTestKit
+  extends FeatureSpecLike with BeforeAndAfterAll with BeforeAndAfterEach with ShouldVerb with Matchers {
+  this: TestKit =>
+
+  import EventSourcedActorTestKit._
+
+  /** defines duration to wait, when expecting events on event log */
+  private val expectEventTimeOut = 100 millisecond
+
+  /** defines how long to wait, when expecting responses in ask patterns. */
+  implicit val askPatternTimeOut = Timeout(FiniteDuration(60, "seconds"))
+
+  // ec for future mapping
+  implicit val ec = system.dispatcher
+
+  /** TestProbe for event log */
+  var logProbe: TestProbe = _
+
+  /** cleans Eventlog before each test */
+  override def beforeEach(): Unit = {
+    logProbe = TestProbe()
+  }
+
+  /** shuts down actor system, once tests are finished */
+  override def afterAll(): Unit = {
+    TestKit.shutdownActorSystem(system)
+  }
+
+  /** get [[ActorRef]] of event log. Must be used as event log for actor's under test. */
+  def eventlog = logProbe.ref
+
+  // ++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+  // +++++++++++ EventsourcedActor initialization +++++++++++
+  // ++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+  /** starts given EventSourcedActor and  returns an [[TestActorRef]] */
+  def initEventSourcedActorAsTestActorRef[T <: EventsourcedActor : ClassTag](creator: ⇒ T) = {
```

Comment:
Rename to `...eventsourced...` to be compliant with all other naming in Eventuate